src/arch.c

```diff
@@ -64,7 +64,7 @@ const struct arch_def *arch_def_native = &arch_def_x86_64;
 const struct arch_def *arch_def_native = &arch_def_arm;
 #elif __aarch64__
 const struct arch_def *arch_def_native = &arch_def_aarch64;
-#elif __loongarch64
+#elif __loongarch__ && __loongarch_grlen == 64
```

Comment:
`__loongarch_lp64`, IMO.  We are caring about the syscall ABI which is an ABI anyway, not the length of the HW register.

Reply:
I briefly looked through the linked reference, and I think @xry111's comment makes sense.  But I will defer to loongarch experts if there's a better answer

Reply:
I think `__loongarch_lp64` should be used here, please refer to this [link](https://loongson.github.io/LoongArch-Documentation/LoongArch-toolchain-conventions-EN.html#_cc_preprocessor_built_in_macro_definitions), Table 14 and Table 15.

Reply:
Okay. I chose to migrate to the `__loongarch_grlen` approach because it's trivial to verify against [the *Toolchain Conventions*](https://loongson.github.io/LoongArch-Documentation/LoongArch-toolchain-conventions-EN.html) that use the same wording. Of course thinking about the actual intent here `__loongarch_lp64` is more appropriate because what we actually care is the ABI; I'll update the branch shortly. Thanks for the reviews!